mazedlx/laravel-feature-policy
# splitup exception methods and introduce contract for public facing methods

## What does this pull request change or introduce
- create separate exception classes from the `InvalidDirective` exception
- introduce a contract for public facing methods

## What is the impact of these changes
- Exception handling

The contract will only add impact when the described methods are altered. Which would be a breaking change.

## Checklist

## Steps to reproduce / test
- run the test suite
- create a new polic
- change the method arguments of the required methods
- an declaration exception for that method should be thrown during a testrun

## Other information:
This change relates to #3
